%3a117/list-queues.sls

```diff
@@ -0,0 +1,17 @@
+(library (srfi :117 list-queues)
+  (export
+    make-list-queue list-queue list-queue-copy list-queue-unfold list-queue-unfold-right
+    list-queue? list-queue-empty?
+    list-queue-front list-queue-back list-queue-list list-queue-first-last
+    list-queue-add-front! list-queue-add-back! list-queue-remove-front! list-queue-remove-back!
+    list-queue-remove-all! list-queue-set-list!
+    list-queue-append list-queue-append! list-queue-concatenate
+    list-queue-map list-queue-map! list-queue-for-each)
+  (import
+    (only (srfi :23) error)
+    (except (rnrs base) error)
+    (rnrs control)
+    (rnrs mutable-pairs)
+    (rnrs records syntactic)
+    (srfi private include))
```

Comment:
Deliberately tried to keep this import set small.  Please advise if there's a better way to accommodate the SRFI-117 reference implementation expecting SRFI-23's (error ...) instead of R6RS (error ...).